src/diagonal.works/b6/graph/graph.go

```diff
@@ -149,40 +149,32 @@ func IsPathUsableByPedestrian(path b6.PathFeature) bool {
 	return false
 }
 
-type ElevationWeights struct{}
+type ElevationWeights struct {
+	UpHillHard   bool
+	DownHillHard bool
+}
 
 func (ElevationWeights) IsUseable(segment b6.Segment) bool {
 	return SimpleHighwayWeights{}.IsUseable(segment)
 }
 
-func (ElevationWeights) Weight(segment b6.Segment) float64 {
-	var weight float64
-	elevation, fromMemory := 0.0, false
-
-	path := segment.Feature
-	for i := 0; i < path.Len()-1; i++ {
-		start := path.Feature(i)
-		stop := path.Feature(i + 1)
-		w := b6.AngleToMeters((*s2.Polyline)(&[]s2.Point{start.Point(), stop.Point()}).Length())
-
-		startElevation, ok := start.Get("ele").FloatValue()
-		if ok {
-			elevation, fromMemory = startElevation, ok
-		} else {
-			startElevation, ok = elevation, fromMemory
-		}
+func (e ElevationWeights) Weight(segment b6.Segment) float64 {
+	weight := b6.AngleToMeters(segment.Polyline().Length())
 
-		stopElevation, ok := stop.Get("ele").FloatValue()
+	startElevation, ok := segment.FirstFeature().Get("ele").FloatValue()
+	if !ok {
+		return weight
+	}
 
-		if fromMemory && ok {
-			if stopElevation > startElevation { // Ascending.
-				// Naismith’s Rule adds ~6s/m of elevation,
-				// which we're normalizing against 1.42m/s avg. walking speed.
-				w += math.Abs(stopElevation-startElevation) * 6 * 1.42
-			}
-		}
+	stopElevation, ok := segment.LastFeature().Get("ele").FloatValue()
+	if !ok {
+		return weight
+	}
 
-		weight += w
+	if stopElevation > startElevation && e.UpHillHard { // Ascending.
```

Comment:
Doesn't this mean that if there's a hill in the middle of the segment (because it's a long segment), then it won't be counted? Don't you need to integrate ascent and descent across the segment?

Reply:
how often is a segment very long? it feels like they are not usually long if you consider a segment the space from node to node? or is it a segment as in a feature in something like osm?

Reply:
not sure about this, reverted that part of logic to submit opts for demo; will come back to it to clean up in the afternoon